zendesk/ticket.go

```diff
@@ -0,0 +1,82 @@
+package zendesk
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type Ticket struct {
+	ID              int64   `json:"id,omitempty"`
+	URL             string  `json:"url,omitempty"`
+	ExternalID      string  `json:"external_id,omitempty"`
+	Type            string  `json:"type,omitempty"`
+	Subject         string  `json:"subject,omitempty"`
+	RawSubject      string  `json:"raw_subject,omitempty"`
+	Description     string  `json:"description,omitempty"`
+	Priority        string  `json:"priority,omitempty"`
+	Status          string  `json:"status,omitempty"`
+	Recipient       string  `json:"recipient,omitempty"`
+	RequesterID     int64   `json:"requester_id"`
+	SubmitterID     int64   `json:"submitter_id,omitempty"`
+	AssigneeID      int64   `json:"assignee_id,omitempty"`
+	OrganizationID  int64   `json:"organization_id,omitempty"`
+	GroupID         int64   `json:"group_id,omitempty"`
+	CollaboratorIDs []int64 `json:"collaborator_ids,omitempty"`
+
+	// TODO:
+	//   Collaborators, type of each array element differs. Any idea to handle this case?
```

Comment:
I have a couple of ideas...
### We go with a slice of interfaces
`collaborators []interface`

### Make a named type  
not much better than slice of interface, but maybe we could make it a named type if that way if we ever wanted some helper methods we would have some options.

Something like:

```go
type Collaborators []interface{}
type Ticket struct {
    collaborators Collaborators
}
```

### We could go with a custom struct and then implement [Marshaler](https://golang.org/pkg/encoding/json/#Marshaler) and [Unmarshaler](https://golang.org/pkg/encoding/json/#Unmarshaler)

Maybe something like this...
```go
type Collaborators struct {
	collaborators []interface{}
}

// Maybe add some append functions
func (c *Collaborators) AppendEmail(email string) {
	c.collaborators = append(c.collaborators, email)
}

func (c *Collaborators) MarshalJSON() ([]byte, error) {
	return json.Marshal(c.collaborators)
}

func (c *Collaborators)  UnmarshalJSON(b []byte) error  {
	var newCollaborators []interface{}
	err :=  json.Unmarshal(b, newCollaborators)
	if err != nil {
		return err
	}
	// possibly validate that there aren't unexpected types in the slice
	c.collaborators = newCollaborators
	return nil
}
```

Reply:
Thank you for your advice. I will try it.

Reply:
put up a draft pr for a potential collaborators type.

https://github.com/nukosuke/go-zendesk/pull/105

Reply:
@tamccall 
I submitted basic ticket implementation on your patch.